tests/integration/records/relationship-changes-test.js

```diff
@@ -949,3 +950,112 @@ test('Calling push with same belongsTo relationship does not trigger observer',
 
   assert.equal(observerCount, 0, 'author observer should not be triggered');
 });
+
+test('Calling push with non empty hasMany relationship does trigger observer', function(assert) {
+  assert.expect(1);
+
+  let observerCount = 0;
+
+  run(function() {
+    let author = env.store.push({
+      data: {
+        id: '2',
+        type: 'author',
+        relationships: {
+          posts: {
+            data: [{ type: 'post', id: '2' }]
+          }
+        }
+      }
+    });
+
+    author.addObserver('posts', function() {
```

Comment:
should be `addObserver('posts.[]')`

the array hasn't been replaced (`posts`), it's been modified (`posts.[]`)